### PR TITLE
Fix drawstyle plot_histogram

### PR DESCRIPTION
**Pull request by ericpre**
### Progress of the PR
- [x] Fix `hs.plot_plot_histograms` drawstyle - see https://matplotlib.org/stable/api/prev_api_changes/api_changes_3.3.0.html#passing-line2d-s-drawstyle-together-with-linestyle-is-removed
- [x] update docstring,
- [N/A] update user guide
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] add tests,
- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
import hyperspy.api as hs
import numpy as np

img = hs.signals.Signal2D(np.random.chisquare(1,[10,10,100]))
img2 = hs.signals.Signal2D(np.random.chisquare(2,[10,10,100]))
ax = hs.plot.plot_histograms([img, img2], legend=['hist1', 'hist2'])
```

### CODE REVIEW

File: hyperspy/drawing/utils.py [hunk 1]
```diff
@@ -1190,12 +1194,18 @@ def plot_spectra(
         For a list, if its length is less than the number of spectra to plot,
         the colors will be cycled. If `None`, use default matplotlib color
         cycle.
-    line_style: {None, matplotlib line style, list of line_styles}, optional
+    line_style : {None, matplotlib line style, list of line_styles}, optional
         Sets the line style of the plots (no action on 'heatmap').
-        The main line style are '-','--','steps','-.',':'.
+        The main line style are ``'-'``, ``'--'``, ``'-.'``, ``':'``.
         For a list, if its length is less than the number of
         spectra to plot, line_style will be cycled. If
```
Comment (jlaehne) [inline hyperspy/drawing/utils.py]: If If

File: hyperspy/drawing/utils.py [hunk 2]
```diff
@@ -1190,12 +1194,18 @@ def plot_spectra(
...
-        If `None`, use continuous lines, eg: ('-','--','steps','-.',':').
+        If `None`, use continuous lines, eg: ``'-'``, ``'--'``, ``'-.'``,
+        ``':'``.
+    drawstyle : {'default', 'steps', 'steps-pre', 'steps-mid', 'steps-post'},
+        default: 'default'
+        The drawstyle determines how the points are connected, no action with
+        ``style='heatmap'``. See matplotlib drawstyle of
+        :py:func:`matplotlib.lines.Line2D` for more information.
     padding : float, optional, default 1.0
```
Comment (jlaehne) [inline hyperspy/drawing/utils.py:1200]: What is the correct style for default value? With or without ':' - this one and the parameter before are inconsistent.

Comment (ericpre) [reply hyperspy/drawing/utils.py:1200]: This is defined in matplotlib, I have tweaked the docstring to make it more clear. Even if this is not obvious from the names,  [`drawstyle`](https://matplotlib.org/stable/api/_as_gen/matplotlib.lines.Line2D.html#matplotlib.lines.Line2D.set_drawstyle) and [`linestyle`](https://matplotlib.org/stable/api/_as_gen/matplotlib.lines.Line2D.html#matplotlib.lines.Line2D.set_linestyle) are two different parameters.
I am wondering, if we should rename `line_style` to `linestyle` to better match with matplotlib [`linestyle`](https://matplotlib.org/stable/api/_as_gen/matplotlib.lines.Line2D.html#matplotlib.lines.Line2D.set_linestyle)?

Comment (jlaehne) [reply hyperspy/drawing/utils.py:1200]: Using the same terminology as matplotlib makes a lot of sense, I would say. So +1  for renaming.

Comment (ericpre) [reply hyperspy/drawing/utils.py:1200]: Done!

Comment (jlaehne) [reply hyperspy/drawing/utils.py:1200]: A few codecov comments have now come up related with the `linestyle` argument. Can you check if it makes sense to add some tests?

Comment (ericpre) [reply hyperspy/drawing/utils.py:1200]: Yes, you are right: there was actually a bug and I simplify a bit the code too. It should be better now!

File: hyperspy/drawing/utils.py [hunk 3]
```diff
@@ -1190,12 +1185,19 @@ def plot_spectra(
...
+    linestyle : {None, matplotlib line style, list of linestyles}, optional
         Sets the line style of the plots (no action on 'heatmap').
-        The main line style are '-','--','steps','-.',':'.
+        The main line style are ``'-'``, ``'--'``, ``'-.'``, ``':'``.
         For a list, if its length is less than the number of
-        spectra to plot, line_style will be cycled. If
-        If `None`, use continuous lines, eg: ('-','--','steps','-.',':').
+        spectra to plot, linestyle will be cycled.
+        If `None`, use continuous lines, eg: ``'-'``, ``'--'``, ``'-.'``,
+        ``':'``.
```
Comment (jlaehne) [inline hyperspy/drawing/utils.py]: Delete `, eg: ``'-'``, ``'--'``, ``'-.'``, ``':'`` `

File: hyperspy/drawing/utils.py [hunk 4]
```diff
@@ -1439,39 +1450,41 @@ def plot_histograms(signal_list,
...
-    optional
-        The main line styles are '-','--','steps','-.',':'.
+    linestyle: {None, valid matplotlib line style, list of line styles}, \
+        optional
+        The main line styles are ``'-'``, ``'--'``, ``'-.'``, ``':'``.
         For a list, if its length is less than the number of
-        spectra to plot, line_style will be cycled.
-        If `None`, use continuous lines, eg: ('-','--','steps','-.',':')
+        spectra to plot, linestyle will be cycled.
+        If `None`, use continuous lines, eg: ``'-'``, ``'--'``, ``'-.'``,
```
Comment (jlaehne) [inline hyperspy/drawing/utils.py]: Delete `, eg: ``'-'``, ``'--'``, ``'-.'``, ``':'`` `

File: hyperspy/drawing/utils.py [hunk 5]
```diff
@@ -1190,12 +1185,19 @@ def plot_spectra(
...
-        The main line style are '-','--','steps','-.',':'.
+        The main line style are ``'-'``, ``'--'``, ``'-.'``, ``':'``.
         For a list, if its length is less than the number of
-        spectra to plot, line_style will be cycled. If
-        If `None`, use continuous lines, eg: ('-','--','steps','-.',':').
+        spectra to plot, linestyle will be cycled.
+        If `None`, use continuous lines, eg: ``'-'``, ``'--'``, ``'-.'``,
+        ``':'``.
+    drawstyle : {'default', 'steps', 'steps-pre', 'steps-mid', 'steps-post'},
+        default: 'default'
```
Comment (jlaehne) [inline hyperspy/drawing/utils.py]: `default 'default'`

**Review by jlaehne — APPROVED** [review]

**Comment by jlaehne** [comment]
A few more corrections to the docstring, otherwise LGTM.

**Pull request merged**